tests/governance_js.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache 2.0 License.
+import infra.network
+import infra.path
+import infra.proc
+import infra.net
+import infra.e2e_args
+import suite.test_requirements as reqs
+
+
+def action(name, **args):
+    return {"name": name, "args": args}
+
+
+@reqs.description("Test proposal validation")
+def test_proposal_validation(network, args):
+    primary, _ = network.find_nodes()
+    add_member = [action("add_member", cert="", enc_pubk="", member_data={})]
```

Comment:
Should be wrapped in `"actions": ...` but probably best to get the PR in and do any further work iteratively.

Reply:
So let's talk about this, but wrapping is awkward as discussed this morning if we think that being precise about the proposal is important (and that's still my opinion) because it implies partial parsing/serialisation.

Reply:
Ok, so here's a suggestion:

Proposals _are_ {"actions": []}, and are stored as such in `public:ccf.gov.proposals.js` (all `.js` suffixes are temporary for disambiguation where necessary). That adds a bit to the validation in the constitution, but it's not too bad.

Then there are:

`GET /gov/proposals.js/{id}/proposal`, which returns the proposal verbatim, `GET /gov/proposals.js/{id}/state`, which returns the contents of `public:ccf.gov.proposals_info`, ie. proposer id, state and votes.

This eliminates the need for partial serialisation/deserialisation.

Reply:
I would probably make it `GET /gov/proposals.js/{id}` which has the proposer id and state (and maybe vote count), and then `GET /gov/proposals.js/{id}/actions` which returns what was submitted, and `GET /gov/proposals.js/{id}/votes` for the votes.

Reply:
Ok, I did nearly that, except I kept the ballots in the first endpoint for now. I think we can default to returning a count and having a url param to ask for expansion.

Reply:
Works for me.